### PR TITLE
feat(search-with-typeahead): introducing component

### PR DESCRIPTION
**Pull request by IgnacioBecerra**
### Related Ticket(s)
#6143 
#6144

### Description
The current Masthead uses Redux actions in order to incorporate the SearchAPI functionality into `dds-masthead-search`. This includes having a container, composite, and a connect component to render the Masthead Search component, as well as including these actions in the Masthead container and composite components. 

This PR extracts all the functionality from `<dds-masthead-search>` and ensures all the Search API and suggestions functionality can work on its own, meaning that we now effectively have a Search component with Typeahead functionality. 

This PR also replaces the aforementioned Masthead Search components within the actual Masthead. 

### Changelog

**New**

- extracted DDSMastheadSearch functionality into its own component DDSSearchWithTypeahead
- DDSSearchWIthTypeahead React wrapper

**Changed**

- replaced all the DDSMastheadSearch components with the new DDSSearchWithTypeahead component from the Masthead
- refactored some of the search styles

**Removed**

- DDSMastheadSearch, Connect, Container, Composite files

### CODE REVIEW

**Comment by jeffchew** [comment]
Thank you @IgnacioBecerra !

I think something happened with the search icon alignment: 

image

**Comment by jeffchew** [comment]
@IgnacioBecerra can you add a cdn.html example?

image

File: packages/web-components/src/components/search-with-typeahead/__stories__/README.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+> [CodeSandbox](https://githubbox.com/carbon-design-system/carbon-for-ibm-dotcom/tree/master/packages/web-components/examples/codesandbox/components/search-with-typeahead)
+> example implementation.
+
+[![Edit @carbon/ibmdotcom-web-components](https://codesandbox.io/static/img/play-codesandbox.svg)](https://githubbox.com/carbon-design-system/carbon-for-ibm-dotcom/tree/master/packages/web-components/examples/codesandbox/components/search-with-typeahead)
+
+## Getting started
+
+Here's a quick example to get you started.
+
+## Example using imports
```
Comment (jeffchew) [inline packages/web-components/src/components/search-with-typeahead/__stories__/README.stories.mdx:19]: This is missing the CDN documentation.

File: packages/web-components/src/components/search-with-typeahead/__stories__/README.stories.react.mdx [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import { Preview, Props, Description, Story } from '@storybook/addon-docs/blocks';
+import contributing from '../../../../../../docs/contributing-license.md';
+import { PropTypesRef } from '@carbon/ibmdotcom-web-components/es/components-react/search-with-typeahead/search-with-typeahead';
+
+# Tag Group
```
Comment (jeffchew) [inline packages/web-components/src/components/search-with-typeahead/__stories__/README.stories.react.mdx:5]: Tag Group 😄

File: packages/web-components/src/components/search-with-typeahead/__stories__/README.stories.react.mdx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+<Props of={PropTypesRef} />
+
+## Stable selectors
+
+See [our README](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/tree/master/packages/web-components#stable-selectors-for-analytics-and-integratione2e-testing-in-web-components) to see how Web Components selector and `data-autoid` should be used.
+
+| Web Components selector    | Compatibility selector | Description |
+| -------------------------- | ---------------------- | ----------- |
+| `<DDSSearchWithTypeahead>` | N/A                    | Component   |
```
Comment (jeffchew) [inline packages/web-components/src/components/search-with-typeahead/__stories__/README.stories.react.mdx:49]: There's no compatibility selector that can be used?

**Comment by jeffchew** [comment]
Hey @IgnacioBecerra , the search opened onload seems off:

image

File: packages/web-components/src/components/masthead/__stories__/masthead.stories.ts [hunk 4]
```diff
@@ -92,7 +91,7 @@ export const WithCustomNavigation = ({ parameters }) => {
     <style>
       ${styles}
     </style>
-    <dds-masthead-container-search-only
+    <dds-masthead-container
```
Comment (annawen1) [inline packages/web-components/src/components/masthead/__stories__/masthead.stories.ts:94]: @IgnacioBecerra if we remove `dds-masthead-container-search-only` and set it to just `dds-masthead-container`, the Custom Navigation menu items get overwritten from the data in the translation files. Let's switch back to `dds-masthead-composite`?

**Comment by shixiedesign** [comment]
Hey @IgnacioBecerra ! Looking pretty good. I'm still checking with designers to see if there are some parts of the spec that's out of date, but here are a few immediate issues to look into: 

1. drop shadow is too strong. Expecting `X:0; Y:0; Blur: 6px; Alpha: 0.3; Color: black`
 
image

2. Focus state is a bit unusual. So seems like it's possible to have multiple focus states. Usually I'd expect only one. When typeahead is open, the focus should go into the dropdown, and on tab, if it leaves dropdown, I expect the dropdown option's blue focus outline to disappear. Currently, it doesn't, so resulting in 2 focused elements: 

Aug-03-2021 13-24-30

**Expecting:** 

Aug-03-2021 13-28-56

**Comment by IgnacioBecerra** [comment]
@shixiedesign Thanks for the feedback! Since the #6143 issue was mainly about extracting the Masthead Search into its own component, I'm not including the Leadspace Search variation in this PR, but rather in the other PR that you mentioned to have everything pertaining to that component in its own.

Latest commit addresses your comments!

**Review by shixiedesign — APPROVED** [review]
Thanks! Looks good to me 👍

**Comment by jeffchew** [comment]
@IgnacioBecerra can you add cypress tests in `/packages/web-components/tests/e2e/__tests__`?

**Comment by jeffchew** [comment]
@IgnacioBecerra the CDN example looks off as well:

image

**Review by jeffchew — APPROVED** [review]
Thank you @IgnacioBecerra !

**Pull request merged**